dace/symbolic.py

```diff
@@ -596,7 +598,7 @@ def pystr_to_symbolic(expr, symbol_map=None, simplify=None):
     locals.update(sympy.abc._clash)
 
     # Sympy processes "not" as direct evaluation rather than negation
-    if isinstance(expr, str) and 'not' in expr:
+    if isinstance(expr, str) and 'not ' in expr:
```

Comment:
`re.search(r"\\bnot\\b", expr) is not None`

Reply:
Fixed